src/pkg/auditext/event/login/login.go

```diff
@@ -0,0 +1,94 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package login
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"regexp"
+	"time"
+
+	"github.com/goharbor/harbor/src/common/rbac"
+	event2 "github.com/goharbor/harbor/src/controller/event"
+	"github.com/goharbor/harbor/src/controller/event/metadata/commonevent"
+	"github.com/goharbor/harbor/src/controller/event/model"
+	"github.com/goharbor/harbor/src/lib/config"
+	"github.com/goharbor/harbor/src/pkg/notifier/event"
+)
+
+func init() {
+	var loginResolver = &resolver{}
+	commonevent.RegisterResolver(`/c/login$`, loginResolver)
+	commonevent.RegisterResolver(`/c/log_out$`, loginResolver)
+}
+
+const (
+	opLogout       = "logout"
+	opLogin        = "login"
+	logoutSuffix   = "log_out"
+	payloadPattern = `principal=(.*?)&password`
+)
+
+type resolver struct {
+}
+
+func (l *resolver) Resolve(ce *commonevent.Metadata, event *event.Event) error {
+	e := &model.CommonEvent{
+		Operator:     ce.Username,
+		ResourceType: rbac.ResourceUser.String(),
+		ResourceName: ce.Username,
+		OcurrAt:      time.Now(),
+	}
+
+	// method POST for login, method GET for logout
```

Comment:
Can we separate this into two resolvers? Making the POST request serve as a login is too restrictive and lacks flexibility.